internal/pkg/override/override.go

```diff
@@ -65,46 +65,28 @@ func (i Info) IsYAMLPatch() bool {
 // If path is a YAML patch document, then IsYAMLPatch evaluates to true.
 // If path is a directory that contains a cdk.json file, then IsCDK evaluates to true.
 func Lookup(path string, fs afero.Fs) (Info, error) {
-	stat, err := fs.Stat(path)
+	_, err := fs.Stat(path)
 	if err != nil {
 		return Info{}, &ErrNotExist{parent: err}
 	}
 
-	if !stat.IsDir() {
-		return lookupYAMLPatch(path, fs)
-	}
-
 	files, err := afero.ReadDir(fs, path)
-	if err != nil {
+	switch {
+	case err != nil:
 		return Info{}, fmt.Errorf("read directory %q: %w", path, err)
-	}
-	var info Info
-	switch n := len(files); n {
-	case 0:
+	case len(files) == 0:
 		return Info{}, fmt.Errorf(`directory at %q is empty`, path)
-	case 1:
-		info, err = lookupYAMLPatch(filepath.Join(path, files[0].Name()), fs)
-		if err != nil {
-			return Info{}, fmt.Errorf("look up YAML patch document when directory contains a single file: %w", err)
-		}
-	default:
-		info, err = lookupCDK(path, fs)
-		if err != nil {
-			return Info{}, fmt.Errorf("look up CDK project for directories with multiple files: %w", err)
-		}
 	}
-	return info, nil
-}
 
-type extension string
+	info, err := lookupYAMLPatch(path, fs)
+	if err == nil { // return yaml info if no error
+		return info, nil
+	}
 
-func (ext extension) isYAML() bool { return ext == ".yml" || ext == ".yaml" }
+	return lookupCDK(path, fs)
```

Comment:
Hmm does that mean if any error occurs when executing `lookupYAMLPatch` we are going to swallow that error? Like users are using yaml patch but the file is misconfigured but we show them error like `"cdk.json" does not exist under...` 🤔

Reply:
:+1: